cmd/main.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+	"os/signal"
+	"weather"
+)
+
+func main() {
+
+	// Config struct
+	config := new(weather.Config)
+
+	// server startup configs
+	flag.IntVar(&config.Port, "port", 9010, "HTTP Server port")
+	flag.StringVar(&config.LogLevel, "log", "verbose", "log level: verbose or quiet")
+
+	flag.Parse()
+
+	s := config.NewServer()
+
+	go func() {
+		err := s.ListenAndServe()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// loop until main route responds
+	weather.WaitForServerRoute(s.Addr + "/weather")
```

Comment:
None of this should be in `main`, should it—it's all paperwork! We want users to be able to do something like:

```go
func main() {
	log.Fatal(api.ListenAndServe())
}
```

Reply:
Got it down to two lines now after watching the functional options vid.

func main() {

	s := weather.NewServer()
	log.Fatal(s.ListenAndServe())

}